### ISSUE TITLE
New sizing handles make it very difficult to move point objects.

### ISSUE THREAD

**Issue by Inverness** [bjorn/tiled#950]
There are several places in my game where I use objects that are only points without a size. Since 0.12.0, moving these have become extremely difficult because the sizing handles cover the entire rectangle. Zooming in gives me enough room to select the object but its still a significant usability issue.

**Comment by bjorn** [comment]
Hmm, this is a bit of an oversight on my part, but I'm not sure what to do to fix it. What you can do is hold the `Alt` modifier, which will make sure a move operation starts rather than a selection operation.

I realize this is a bit hidden, so I wonder a little about alternatives. Suggestions are welcome.

**Comment by Inverness** [comment]
Yes that is a useful thing to know, thanks.

I think the simplest fix is having the sizing handles be positioned relative to the object's shape in the editor instead of its actual shape. That way for point objects they'll be displayed outside of the shape's border just as they would for any other object.

**Comment by bjorn** [comment]
Yes, well, there's a bit of an anomaly going on when objects get reduced to 0 size. A 50x50 object gets rendered as 50x50, a 1x1 objects gets rendered as 1x1, but a 0x0 object gets rendered as a 20x20 rectangle centered on its position. This is a hack that existed just to make such objects workable, and I'm not sure how to extend this hack with the resizing handles. You can't really just place them on the outside of the dummy rectangle.

Previously there was only one handle for the bottom-left corner of the object, and for 0x0 objects this handle would end up in the middle, leaving enough room to grab the 20x20 dummy rectangle. The new group of resize handles takes up way more space. It could be fixed by increasing the size of the dummy rectangle, but then it would need to be quite huge.

**Comment by Inverness** [comment]
>You can't really just place them on the outside of the dummy rectangle.

Why not? That seems like the simplest solution.

If not that, then how about reverting to something like the previous behavior? For a point object there would only be a handle on the bottom right corner. You're only going to need it if you want to expand the point object into something with an area, in which case the other handles will then become available again.

**Comment by bjorn** [comment]
> Why not? That seems like the simplest solution.

Well, because this rectangle is only drawn when the object is 0x0, and it is centered around the position rather than having the position of the object in the top-left. When you would place the handles around it, and the user would start resizing the object, what should happen? As the object would gain some size, the rectangle would shift and reduce in size to match the actual object, and the handles would then either not be in the right place, or they would somehow need to shift along with it.

> If not that, then how about reverting to something like the previous behavior? For a point object there would only be a handle on the bottom right corner. You're only going to need it if you want to expand the point object into something with an area, in which case the other handles will then become available again.

I'm considering whether it may be easier when point objects would be supported as a separate type now. That would make it entirely obvious when those handles should show or not, rather than having a special case once an object reaches 0 size.

**Comment by Inverness** [comment]
I think having a point object be a separate type would make more sense here. It still needs to support rotation though.
